Aliucord/src/main/java/com/aliucord/coreplugins/polls/details/PollDetailsScreen.kt

```diff
@@ -75,18 +77,25 @@ internal class PollDetailsScreen : AppFragment(Utils.getResId("widget_manage_rea
         setActionBarTitle("Poll Votes")
         setActionBarSubtitle(poll.question.text)
 
-        subscription = PollsStore.subscribeOnMain(channelId, messageId) {
-            data = it
-            updateData()
+        subscription = PollsStore.subscribeOnMain(channelId, messageId) { newData ->
+            if (newData == null) { // Message is deleted
+                this.appActivity.finish()
+                return@subscribeOnMain
```

Comment:
Shouldn't you unsubscribe here?

Reply:
The system will call `onDestroy()` for us, where it will get unsubscribed.